// vite-plugin-storybook-nextjs/dist/plugins/next-mocks/alias/link/index.js
import React from 'react';
import { fn } from 'storybook/test';

// src/plugins/next-mocks/alias/link/index.tsx
var linkAction = fn().mockName("next/link::Link");
var MockLink = React.forwardRef(
  function MockLink2({
    href,
    as: _as,
    replace,
    scroll,
    shallow,
    prefetch,
    passHref,
    legacyBehavior,
    locale,
    onClick,
    children,
    ...rest
  }, ref) {
    const hrefString = typeof href === "object" ? `${href.pathname || ""}${href.query ? `?${new URLSearchParams(
      href.query
    ).toString()}` : ""}${href.hash || ""}` : href;
    const handleClick = (e) => {
      e.preventDefault();
      onClick?.(e);
      linkAction(hrefString, { replace, scroll, shallow, prefetch, locale });
    };
    if (legacyBehavior) {
      const child = React.Children.only(children);
      const childProps = {
        ref,
        onClick: (e) => {
          e.preventDefault();
          if (child.props && typeof child.props.onClick === "function") {
            child.props.onClick(e);
          }
          linkAction(hrefString, {
            replace,
            scroll,
            shallow,
            prefetch,
            locale
          });
        },
        ...rest
      };
      if (passHref || child.type === "a" && !("href" in (child.props || {}))) {
        childProps.href = hrefString;
      }
      return React.cloneElement(child, childProps);
    }
    return /* @__PURE__ */ React.createElement("a", { ref, href: hrefString, onClick: handleClick, ...rest }, children);
  }
);
MockLink.displayName = "NextLink";
var link_default = MockLink;
var useLinkStatus = fn(() => ({
  pending: false
})).mockName("next/link::useLinkStatus");

export { MockLink as Link, link_default as default, useLinkStatus };
