// vite-plugin-storybook-nextjs/dist/plugins/next-mocks/alias/link/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var test = require('storybook/test');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React__default = /*#__PURE__*/_interopDefault(React);

// src/plugins/next-mocks/alias/link/index.tsx
var linkAction = test.fn().mockName("next/link::Link");
var MockLink = React__default.default.forwardRef(
  function MockLink2({
    href,
    as: _as,
    replace,
    scroll,
    shallow,
    prefetch,
    passHref,
    legacyBehavior,
    locale,
    onClick,
    children,
    ...rest
  }, ref) {
    const hrefString = typeof href === "object" ? `${href.pathname || ""}${href.query ? `?${new URLSearchParams(
      href.query
    ).toString()}` : ""}${href.hash || ""}` : href;
    const handleClick = (e) => {
      e.preventDefault();
      onClick?.(e);
      linkAction(hrefString, { replace, scroll, shallow, prefetch, locale });
    };
    if (legacyBehavior) {
      const child = React__default.default.Children.only(children);
      const childProps = {
        ref,
        onClick: (e) => {
          e.preventDefault();
          if (child.props && typeof child.props.onClick === "function") {
            child.props.onClick(e);
          }
          linkAction(hrefString, {
            replace,
            scroll,
            shallow,
            prefetch,
            locale
          });
        },
        ...rest
      };
      if (passHref || child.type === "a" && !("href" in (child.props || {}))) {
        childProps.href = hrefString;
      }
      return React__default.default.cloneElement(child, childProps);
    }
    return /* @__PURE__ */ React__default.default.createElement("a", { ref, href: hrefString, onClick: handleClick, ...rest }, children);
  }
);
MockLink.displayName = "NextLink";
var link_default = MockLink;
var useLinkStatus = test.fn(() => ({
  pending: false
})).mockName("next/link::useLinkStatus");

exports.Link = MockLink;
exports.default = link_default;
exports.useLinkStatus = useLinkStatus;
